docs/modules/listener-operator/pages/overview.adoc

```diff
@@ -0,0 +1,92 @@
+
+[#routing]
+== Request routing
+
+We'll usually want to support running more than one replica of a given service.. which means that we need some way to direct clients to the right server.
+
+=== Server-side request routing
+
+Sometimes, clients should just connect to _any_ instance of the service, and the service itself is responsible for Doing The Right Thing(tm).
+
+This is common for stateless front-ends, or for services that handle a more complicated consensus protocol internally (like ZooKeeper).
+
+Kubernetes traditionally handles this through deploying a common https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#service-v1-core[Service] across the entire replica set.
+
+The Listener operator supports this by manually deploying a common xref:listener.adoc[] object, and then xref:volume.adoc#shared-exposure[mounting] it into each replica.
+
+NOTE: However, these listeners _may_ still have more than one address (for example: when using xref:listenerclass.adoc#servicetype-nodeport[node-bound listeners]). Clients are responsible for picking a random target address from the options given.
+
+=== Client-side request routing
+
+Sometimes, the client needs to connect to a specific instance of a service.
+
+For example, a HDFS client connects to _all_ available NameNodes to find the current primary instance, then queries _that_ one for which DataNode has the file that it is looking for.
+
+Kubernetes doesn't really handle this natively, you would need to deploy a separate Service for each
+
+The Listener operator supports this by automatically creating xref:listener.adoc[Listeners] corresponding to each xref:volume.adoc[] when xref:volume.adoc#individual-pod-exposure[requested].
+
+[#address-stability]
+== Address stability
+
+We want to avoid clients needing to update their connection configuration just because the service got redeployed.
+
+The Listener operator binds the lifetime of an address to the lifetime of the xref:volume.adoc[Listener Volume].
+As long as the PersistentVolume exists, the xref:listener.adoc[] is expected to keep the same address.
+If the PersistentVolume(Claim) is deleted (and recreated), then the address may{empty}footnote:[But isn't always.] be changed.
+Long-lived bindings can be created through `StatefulSet.spec.volumeClaimTemplates`, which creates "permanent" PersistentVolumes, which are not cleaned up automatically and must be deleted manually by an administrator once they are no longer used.
+Short-lived bindings should be created through `Pod.spec.volumes.ephemeral`, which creates a PersistentVolume that will automatically be deleted once the Pod no longer exists.
+
+WARNING: Ephemeral CSI volumes (configured via `Pod.spec.volumes.csi`) are an entirely different thing entirely, and are not supported by the Listener operator. Ephemeral PersistentVolumeClaims (`.volumes.ephemeral`) are still "Persistent" from CSI's point of view.
+
+The listener operator provides two tools for dealing with this:
```

Comment:
```suggestion
The listener operator provides three tools for dealing with this:
```

Looks like three possiblities?

Reply:
a9b4362a9e3d151ebde11be47b17d24320100a90

I'd consider "internal" and "external" to be subvariants of the "load balancer" tool, but I changed it to "a few" instead because this is the sort of thing that easily goes out of date in the future.